tests/lib/ocpp/v201/test_database_handler.cpp

```diff
@@ -65,4 +66,91 @@ TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithProfileData_InsertNewProfile) {
     ASSERT_EQ(count, 2);
 }
 
+TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithProfileData_DeleteRemovesSpecifiedProfiles) {
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{.id = 1, .stackLevel = 1});
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{.id = 2, .stackLevel = 1});
+
+    auto sql = "SELECT COUNT(*) FROM CHARGING_PROFILES";
+
+    auto select_stmt = db_connection->new_statement(sql);
+
+    do {
```

Comment:
Opportunity to refactor into a helper function to remove duplication?

Reply:
In a normal circumstance yes I would refactor this. However, there is a way that this can be done without needing this loop and I want to call it out to the community. This is just depicting the issue with the current interface.